### PR TITLE
Supporting source file launcher in VS Code extension

### PR DESCRIPTION
**Pull request by lahodaj**
(Single) source file launcher support has multiple problems inside VS Code - especially settings that are inside NetBeans done through file attributes are not accessible. This patch attempts to use the Run Configuration tab/settings to provide options for launching the single file. The same options are also parsed for use in the editor.

---
**^Add meaningful description above**

By opening a pull request you confirm that, unless explicitly stated otherwise, the changes -

 - are all your own work, and you have the right to contribute them.
 - are contributed solely under the terms and conditions of the Apache License 2.0 (see section 5 of the license for more information).

Please make sure (eg. `git log`) that all commits have a valid name and email address for you in the Author field.

If you're a first time contributor, see the Contributing guidelines for more information.

If you're a committer, please label the PR before pressing "Create pull request" so that the right test jobs can run.

### CODE REVIEW

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/singlesourcefile/CompilerOptionsQueryImpl.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+                    continue;
+                String parameter = newOptions.get(i + 1);
+                switch (newOptions.get(i)) {
+                    case "-classpath": case "-cp": case "--class-path":
+                        classpath = parameter;
+                        break;
+                    case "--module-path": case "-p":
+                        modulepath = parameter;
+                        break;
+                    case "--source":
```
[the remark this answers is not in the document]
Comment (lahodaj) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/singlesourcefile/CompilerOptionsQueryImpl.java:224]: Thanks, but I don't think it should be here - these are source launcher options (as opposed to javac options), and source launcher does not support `-source`.

**Comment by lahodaj** [comment]
@sdedic thanks a lot for the comments!. I've mostly (except the `-source`) reflected them under [bc9417f](https://github.com/apache/netbeans/pull/6262/commits/bc9417f46a31efa22c6827f07c71a1e5b1a8c864).

**Comment by arvindaprameya** [comment]
This looks good, thank you for making this change! Appoved

**Review by sdedic — APPROVED** [review]

**Review by dbalek — APPROVED** [review]
Looks fine. Thanks.

**Pull request merged**